PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -364,6 +364,17 @@ def toolCode(self, process):
         #pre-preparation to run over miniAOD 
         if onMiniAOD:            
             self.miniAODConfigurationPre(process, patMetModuleSequence, pfCandCollection, postfix)
+        else:
+            from PhysicsTools.PatUtils.pfeGammaToCandidate_cfi import pfeGammaToCandidate
+            task = getPatAlgosToolsTask(process)
+            addToProcessAndTask("pfeGammaToCandidate", pfeGammaToCandidate.clone(
+                                  electrons = electronCollection,
+                                  photons = photonCollection),
```

Comment:
based on https://github.com/cms-sw/cmssw/pull/29385#discussion_r406874715
do I understand correctly that this is the origin of the problem (a "reference" is passed instead of a literal/value)? In all other cases a value is passed.
If so, shouldn't this be changed to ```= electronCollection.value()``` or ``` = getValue("electronCollection")``` 
or copy.copy or cms.InputTag(

Reply:
The best way to think about it is all variables in python are 'references'. Python really does not have the idea of 'pass by value'.

Reply:
right, I think I knew this before this PR was made, but then the problem was obfuscated a bit and it began to look like even unnamed "xyz" mentioned on different lines in assignments to different variables can couple those different variables.